tests/dataflow/test_unified_systolic.py

```diff
@@ -0,0 +1,121 @@
+# Copyright Allo authors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import allo
+from allo.ir.types import int8, int32, bool, Stream
+import allo.dataflow as df
+import allo.backend.hls as hls
+import numpy as np
+import pytest
+
+M, N, K = 4, 4, 4
+Rt, Ct = 2, 2
+# Rt, Ct = 4, 4
+# M, N, K = 16, 16, 16
+# Rt, Ct = 4, 4
+# Rt, Ct = 16, 16
+P0, P1 = Rt + 2, Ct + 2
+
+
+@df.kernel(mapping=[P0, P1])
+def gemm(A: int32[M, K], B: int32[K, N], inst: int8[2], C: int32[M, N]):
```

Comment:
Can you pull the latest commit (need to rebuild) and see whether a single bool inst now works for both LLVM and HLS backend? If so, you can use `inst: bool`.

Reply:
Sure. I'll try this later.